### PR TITLE
Scope change app

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes: AB#FILL_IN_YOUR_ISSUE_ID
[AB#152262](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/152262)
[AB#152263](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/152263)
[AB#152287](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/152287)
[AB#152288](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/152288)

## Checklist:

-   [x] I have performed a self-review of my own code.
-   [x] I have linked my DevOps task using the AB# tag.
-   [x] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/Core/DataFactory/Components/ConfirmationDialog.tsx [hunk 1]
```diff
@@ -16,32 +16,34 @@ export const ConfirmationDialog = ({
...
-                                onConfirm && onConfirm();
-                            }}
-                        >
-                            Ok
-                        </Button>
-                    </ButtonContainer>
-                </DialogPadding>
-            </Dialog>
+            <Scrim isDismissable={false}>
+                <Dialog style={{ width: 'auto' }}>
```
Comment (Gustav-Eikaas) [inline src/Core/DataFactory/Components/ConfirmationDialog.tsx:20]: style??

File: src/Core/DataFactory/Components/FactoryComponent.tsx [hunk 2]
```diff
@@ -14,49 +14,83 @@ export function FactoryComponent({ onClose }: FactoryComponentProps): JSX.Elemen
...
+        if (hasUnsavedChanges) {
+            setShowDialog(true);
+        } else {
+            close();
+        }
+    };
+
+    const close = () => {
+        clearActiveFactory();
+        onClose && onClose;
```
Comment (Gustav-Eikaas) [inline src/Core/DataFactory/Components/FactoryComponent.tsx:29]: ()

File: src/Core/DataFactory/Components/FactoryComponent.tsx [hunk 3]
```diff
@@ -14,49 +14,83 @@ export function FactoryComponent({ onClose }: FactoryComponentProps): JSX.Elemen
...
+                        }
                     }}
                     isDismissable={true}
                 >
-                    <Container>
-                        {showDialog && (
-                            <Scrim isDismissable={false}>
+                    <ScrimContainer>
+                        <Container>
+                            {showDialog && (
```
Comment (Gustav-Eikaas) [inline src/Core/DataFactory/Components/FactoryComponent.tsx:49]: TODO add todo refactor global dialog

File: src/apps/ScopeChangeRequest/Components/CustomSidesheet.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                <>
+                    <Wrapper>
+                        <TitleHeader>
+                            <Field
+                                label={'Review scope change request'}
+                                value=""
+                                customLabel={{ fontSize: 'xx-large' }}
+                            />
+                            <Button variant="ghost_icon" onClick={onClose}>
+                                <h2>x</h2>
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/CustomSidesheet.tsx:29]: Close icon

File: src/apps/ScopeChangeRequest/Components/WorkflowEditor/Components/WorkflowEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+const ButtonPair = styled.div`
+    display: flex;
+    flex-direction: row;
+`;
+
+const HorizontalSpacer = styled.div`
+    margin: 5px;
+`;
+
+const WorkflowStepsOutline = styled.div``;
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/WorkflowEditor/Components/WorkflowEditor.tsx:175]: remove

File: src/apps/apps.ts [hunk 6]
```diff
@@ -157,6 +159,10 @@ export const apps: AppManifest[] = [
         groupe: Apps.ProjectInformation,
         icon: '',
         uri: '',
+        app: {
```
Comment (Gustav-Eikaas) [inline src/apps/apps.ts:162]: remove

File: src/components/CompletionView/src/Context/DataProvider.tsx [hunk 7]
```diff
@@ -18,7 +20,7 @@ interface DataState {
     name: string;
     data: any[];
     subData: Record<string, any[]>;
-    itemId: string;
+    item: any;
```
Comment (Gustav-Eikaas) [inline src/components/CompletionView/src/Context/DataProvider.tsx:23]: convert to Record<string, unknown>;

File: src/components/CompletionView/src/Context/DataProvider.tsx [hunk 8]
```diff
@@ -113,10 +121,10 @@ export const DataProvider = ({ children }: DataProviderProps): JSX.Element => {
             console.warn(`Data may not be valid. Data validator is not registered for ${name}.`);
             dispatch(actions.getData(data));
         }
-    }, [dataFetcher, name, validator]);
+    }, []);
 
-    const setSelected = (itemId: string) => {
-        dispatch(actions.setSelectedItem(itemId !== state.itemId ? itemId : ''));
+    const setSelected = (item: any) => {
+        dispatch(actions.setSelectedItem(item));
```
Comment (Gustav-Eikaas) [inline src/components/CompletionView/src/Context/DataProvider.tsx:127]: itemId !== state.itemId ? itemId : '')

File: src/components/CompletionView/src/DataViewerApi/DataViewState.ts [hunk 9]
```diff
@@ -84,9 +85,17 @@ export interface PowerBiOptions {
 
 export type StatusFunc<T> = (data: T[]) => StatusItem[];
 
+export interface DataViewSideSheetOptions<T> {
+    CustomRender?: React.FC<{ item: T; onClose: () => void }>;
```
Comment (Gustav-Eikaas) [inline src/components/CompletionView/src/DataViewerApi/DataViewState.ts:89]: CustomComponent

File: src/components/CompletionView/src/Tabs/tabsConfig.ts [hunk 10]
```diff
@@ -49,6 +51,11 @@ const tabsConfig: TabsConfigItem[] = [
         icon: AnalyticsIcon,
         viewComponent: PowerBiTab,
     },
+    {
+        title: 'Editor',
+        icon: GantIcon,
```
Comment (Gustav-Eikaas) [inline src/components/CompletionView/src/Tabs/tabsConfig.ts:56]: add todo

**Review by Noggling — APPROVED** [review]

File: src/Core/DataFactory/Components/ConfirmationDialog.tsx (hunk 1 above)
Comment (Gustav-Eikaas) [reply src/Core/DataFactory/Components/ConfirmationDialog.tsx:20]: Has to be like this

**Pull request merged**